test/Microsoft.DotNet.MSBuildSdkResolver.Tests/GivenAnMSBuildSdkResolver.cs

```diff
@@ -106,11 +108,59 @@ public void ItReturnsNullWhenTheSDKRequiresAHigherVersionOfMSBuildThanTheOneAvai
         }
 
         [Fact]
-        public void ItReturnsTheVersionIfItIsEqualToTheMinVersion()
+        public void ItReturnsNullWhenTheDefaultVSRequiredSDKVersionIsHigherThanTheSDKVersionAvailable()
+        {
+            var environment = new TestEnvironment();
+            var expected =
+                environment.CreateSdkDirectory(ProgramFiles.X64, "Some.Test.Sdk", "1.0.1");
+            environment.CreateMuxerAndAddToPath(ProgramFiles.X64);
+
+            var resolver = environment.CreateResolver();
```

Comment:
No specific to this PR. but the testing strategy of this part is strange.

We have a mix of abstraction level, environment.CreateSdkDirectory should eventually create a real directory in the file system from reading the code. However, we also has a MockContext that does a monkey patch in memory.

Casting to (MockResult) is also potentially flaky if the object changed in SDK repo, and it will be an cryptic runtime error for this test.